Jenkinsfile

```
@@ -1,31 +1,6 @@
-#!groovy
+#!/usr/bin/env groovy
 
-def deployBranches = [ "master" ]
-def phase = "verify"
-
-stage ('Build') {
-    node {
-        checkout scm
-        def branch = scm.branches[0].name
-        if (deployBranches.contains(branch)) {
-            phase = "deploy"
-        }
-        echo "Running mvn $phase on branch $branch"
-        sh 'mkdir -p ~/.gnupg'
-        withCredentials([
-            file(credentialsId: 'gpg-pubring', variable: 'GPG_PUB_RING'),
-            file(credentialsId: 'gpg-secring', variable: 'GPG_SEC_RING'),
-            file(credentialsId: 'gradle-settings', variable: 'GRADLE_SETTINGS')]) {
-                try {
-                    sh "./gradlew $phase -P signing.secretKeyRingFile=$GPG_SEC_RING -P extProps=$GRADLE_SETTINGS"
-                } finally {
-                    archiveArtifacts 'build/libs/*.jar'
-                    archiveArtifacts 'build/libs/*.asc'
-                    if (phase == 'deploy') archiveArtifacts 'build/poms/*.xml'
-                    if (phase == 'deploy') archiveArtifacts 'build/poms/*.asc'
-                    junit allowEmptyResults: true, testResults: 'build/test-results/test/*.xml'
-                }
-        }
-        step([$class: 'WsCleanup'])
-    }
-}
\ No newline at end of file
+parallel(failFast: true,
+    'Linux Build': { node('linux') { buildPluginWithGradle() } },
+    'Windows Build': { node('windows') { buildPluginWithGradle() } },
+)
```

you should just need:
```suggestion
buildPluginWithGradle(configurations: [
  [ platform: 'linux', jdk: '8' ],
  [ platform: 'linux', jdk: '11' ],
  [ platform: 'windows', jdk: '11' ]
])
```

---

I'll add a row for JDK8 on Windows, just to be complete.